### PR TITLE
Add const to input_fmt

### PR DESCRIPTION
**Pull request by jordemort**
Hey folks, ran into a compile error building this against ffmpeg 5.0 - the compiler really wanted the input_fmt to be const.

### CODE REVIEW

**Comment by naushir** [comment]
This is a bit of an annoying one.  Our current ffmpeg build v4.3 requires a non-const value to ``avformat_open_input``:
```
int avformat_open_input(AVFormatContext **ps, const char *url, ff_const59 AVInputFormat *fmt, AVDictionary **options);
```
so this change will break existing builds.  But as you noted, for ffmpeg v5.0 and later this will have to change to a const.

Unfortunately, I don't think we can take this change as-is yet as we are not ready to move to ffmpeg v5.0 in our releases just yet.

**Comment by naushir** [comment]
You can see the compile error from the CI build [here](https://github.com/raspberrypi/libcamera-apps/runs/7687565735?check_suite_focus=true).

File: encoder/libav_encoder.cpp [hunk 1]
```diff
@@ -129,7 +129,7 @@ void LibAvEncoder::initVideoCodec(VideoOptions const *options, StreamInfo const
 
 void LibAvEncoder::initAudioInCodec(VideoOptions const *options, StreamInfo const &info)
 {
-	AVInputFormat *input_fmt = (AVInputFormat *)av_find_input_format("pulse");
+	const AVInputFormat *input_fmt = (AVInputFormat *)av_find_input_format("pulse");
```
Comment (naushir) [inline encoder/libav_encoder.cpp:133]: Perhaps things would work for all versions if you changed ``const`` to ``ff_const59``?

Comment (naushir) [reply encoder/libav_encoder.cpp:133]: Probably not - see avformat.h:

```
    /**
     * The ff_const59 define is not part of the public API and will
     * be removed without further warning.
     */
```

Comment (naushir) [reply encoder/libav_encoder.cpp:133]: You may be able to use the ``FF_API_AVIOFORMAT`` define, It's not clear what exactly that is meant to be used for, but might work for our problem here.

Comment (jordemort) [reply encoder/libav_encoder.cpp:137]: It sounds like FF_API_AVIOFORMAT is also private, re: https://www.mail-archive.com/ffmpeg-devel@ffmpeg.org/msg81099.html

> Could you add a comment stating that, much like the FF_API_ defines, this ff_const59 define is not part of the public API and will eventually disappear without warning?

It seems like LIBAVUTIL_VERSION_MAJOR is intended for public consumption tho, so I've given that a shot.

Comment (naushir) [reply encoder/libav_encoder.cpp:137]: Looks good.  Once the checks have passed, I'll merge this in.  Thanks!

**Comment by naushir** [comment]
Actually, would you be able to re-submit this as a single patch instead of 2 commits?  Would be tidier in the history.

File: encoder/libav_encoder.cpp (hunk 1 above)
Comment (jordemort) [reply encoder/libav_encoder.cpp:137]: I'm not sure if it's correct but it passes your CI and also builds with 5.0 over here :)

**Comment by jordemort** [comment]
> Actually, would you be able to re-submit this as a single patch instead of 2 commits? Would be tidier in the history.

Squashed down to one commit, I tried to write a better commit message for you too

**Comment by naushir** [comment]
Great, I'll merge this as soon as the tests have completed.

**Pull request merged**